stumpy/mmotifs.py

```diff
@@ -0,0 +1,202 @@
+        The p-norm to apply for computing the Minkowski distance. This parameter is
+        ignored when `normalize == False`.
+
+    Returns
+    -------
+    motif_distances: numpy.ndarray
+        The distances corresponding to a set of subsequence matches for each motif.
+
+    motif_indices: numpy.ndarray
+        The indices corresponding to a set of subsequences matches for each motif.
+
+    motif_subspaces: list
+        A list consisting of arrays that contain the `k`-dimensional
+        subspace for each motif.
+
+    motif_mdls: list
+        A list consisting of arrays that contain the mdl results for
+        finding the dimension of each motif
+    """
+    T = core._preprocess(T)
+    m = T.shape[-1] - P.shape[-1] + 1
+
+    if max_motifs < 1:  # pragma: no cover
+        logger.warning(
+            "The maximum number of motifs, `max_motifs`, "
+            "must be greater than or equal to 1"
+        )
+        logger.warning("`max_motifs` has been set to `1`")
+        max_motifs = 1
+
+    T, M_T, Σ_T = core.preprocess(T, m)
+
+    excl_zone = int(np.ceil(m / config.STUMPY_EXCL_ZONE_DENOM))
+
+    if max_matches is None:
+        max_matches = np.inf
+
+    if cutoffs is None:
+        P_copy = P.copy().astype(np.float64)
+        P_copy[np.isinf(P_copy)] = np.nan
+        cutoffs = np.nanmax(
+            [
+                np.nanmean(P_copy, axis=1) - 2.0 * np.nanstd(P_copy, axis=1),
+                np.nanmin(P_copy, axis=1),
+            ],
+            axis=0,
+        )
+    if np.isscalar(cutoffs):
+        cutoffs = np.full(P.shape[0], cutoffs)
+
+    motif_distances = []
+    motif_indices = []
+    motif_subspaces = []
+    motif_mdls = []
+
+    candidate_idx = np.argmin(P, axis=1)
+    nn_idx = I[np.arange(len(candidate_idx)), candidate_idx]
+
+    while len(motif_distances) < max_motifs:
+        if k_s is None:
+            mdls, subspaces = mdl(T, m, candidate_idx, nn_idx, include)
+            k = np.argmin(mdls)
+            sub = subspaces[k]
+        if np.isscalar(k_s):
+            k = k_s
+
+        motif_idx = candidate_idx[k]
+        motif_value = P[k, motif_idx]
+
+        if np.isscalar(k_s):
+            sub = subspace(T, m, motif_idx, nn_idx[k], k, include)
+
+        if (
+            motif_value > cutoffs[k]
+            or not np.isfinite(motif_value)
+            or (isinstance(max_distance, float) and motif_value > max_distance)
+        ):
+            break
+
+        query_matches = match(
+            Q=T[sub, motif_idx : motif_idx + m],
+            T=T[sub],
+            M_T=M_T[sub],
+            Σ_T=Σ_T[sub],
+            max_matches=max_matches,
+            max_distance=max_distance,
+            atol=atol,
+            normalize=normalize,
+            p=p,
+        )
+
+        if len(query_matches) > min_neighbors:
+            motif_distances.append(query_matches[:, 0])
+            motif_indices.append(query_matches[:, 1])
+            if k_s is None:

```

Comment:
This `if/else` section feels really weird and, given the comments above, we can just go back to `motif_mdls.append(mdls)` and remove the `if/else`